src/managers/quorum.py

```diff
@@ -102,17 +103,19 @@ def get_hostname_mapping(self) -> dict[str, str]:
 
         return {"hostname": hostname, "fqdn": fqdn, "ip": ip}
 
-    def _get_updated_servers(self, add: list[str], remove: list[str]) -> dict[str, str]:
+    def _get_updated_servers(self, add: list[str]) -> dict[str, str]:
         """Simple wrapper for building `updated_servers` for passing to app data updates."""
-        servers_to_update = add + remove
-
         updated_servers = {}
-        for server_string in servers_to_update:
+        for server_string in add:
             unit_id = str(int(re.findall(r"server.([0-9]+)", server_string)[0]) - 1)
             if server_string in add:
                 updated_servers[unit_id] = "added"
-            elif server_string in remove:
-                updated_servers[unit_id] = "removed"
+
+        # settings units to removed that were handled in relation-departed
+        # also set here in case leader missed the event
+        for added_id in self.state.cluster.added_unit_ids:
+            if added_id not in [server.unit_id for server in self.state.servers]:
+                updated_servers[str(added_id)] = "removed"
```

Comment:
**question** Why we don't do this in the relation-departed event? Here it seems logically a bit out of place

Reply:
I think this is related to this:
> If the Juju leader IS departing, we might miss the events to set peer-data, but will 'eventually' be cleaned up by the newly elected leader during a reconciliation run of update-status --> update_quorum

Reply:
Good idea, will do.